sklearnex/dispatcher.py

```diff
@@ -1,225 +1,277 @@
+# limitations under the License.
+# ===============================================================================
+
+import os
+
+# Other imports
+import sys
+from functools import lru_cache
+
+from daal4py.sklearn._utils import daal_check_version, sklearn_check_version
+
+
+def _is_new_patching_available():
+    return os.environ.get("OFF_ONEDAL_IFACE") is None and daal_check_version(
+        (2021, "P", 300)
+    )
+
+
+def _is_preview_enabled():
+    return os.environ.get("SKLEARNEX_PREVIEW") is not None
+
+
+@lru_cache(maxsize=None)
+def get_patch_map():
+    from daal4py.sklearn.monkeypatch.dispatcher import _get_map_of_algorithms
+
+    mapping = _get_map_of_algorithms().copy()
+
+    if _is_new_patching_available():
+        # Classes for patching
+
+        # Scikit-learn* modules
+        import sklearn as base_module
+        import sklearn.decomposition as decomposition_module
+        import sklearn.ensemble as ensemble_module
+        import sklearn.linear_model as linear_model_module
+        import sklearn.neighbors as neighbors_module
+        import sklearn.svm as svm_module
+
+        from ._config import config_context as config_context_sklearnex
+        from ._config import get_config as get_config_sklearnex
+        from ._config import set_config as set_config_sklearnex
+        from .neighbors import KNeighborsClassifier as KNeighborsClassifier_sklearnex
+        from .neighbors import KNeighborsRegressor as KNeighborsRegressor_sklearnex
+        from .neighbors import LocalOutlierFactor as LocalOutlierFactor_sklearnex
+        from .neighbors import NearestNeighbors as NearestNeighbors_sklearnex
+
+        # Preview classes for patching
+        from .preview.decomposition import PCA as PCA_sklearnex
+        from .preview.ensemble import (
+            RandomForestClassifier as RandomForestClassifier_sklearnex,
+        )
+        from .preview.ensemble import (
+            RandomForestRegressor as RandomForestRegressor_sklearnex,
+        )
+        from .preview.linear_model import LinearRegression as LinearRegression_sklearnex
+        from .svm import SVC as SVC_sklearnex
+        from .svm import SVR as SVR_sklearnex
+        from .svm import NuSVC as NuSVC_sklearnex
+        from .svm import NuSVR as NuSVR_sklearnex
+
+        # Patch for mapping
+        # Algorithms
+
+        if _is_preview_enabled():
+            # Ensemble
+            mapping.pop("random_forest_classifier")
+            mapping.pop("random_forest_regressor")
+            mapping.pop("randomrorestclassifier")
+            mapping.pop("randomforestregressor")
+            mapping["random_forest_classifier"] = [
+                [
+                    (
+                        ensemble_module,
+                        "RandomForestClassifier",
+                        RandomForestClassifier_sklearnex,
+                    ),
+                    None,
+                ]
+            ]
```

Comment:
Is there any better non-sparse formatting for this mapping entry like this:
```suggestion
            mapping["random_forest_classifier"] = [[(
                ensemble_module,
                "RandomForestClassifier",
                RandomForestClassifier_sklearnex,
            ), None]]
```

Reply:
I think stuff like this should be handled in a `pyproject.toml` which we don't yet have
See for example https://github.com/psf/black/blob/main/pyproject.toml#L8